content/reference/wadm/api.md

```diff
@@ -0,0 +1,218 @@
+---
+title: "API Reference"
+date: 2018-12-29T11:02:05+06:00
+weight: 2
+draft: false
+description: "wadm API Specification"
+---
+
+The normal way to interact with a `wadm` installation (which could be a single server or a cluster) is through the [wash](../../wash) command-line tool. However, if you are planning on creating your own integration or writing a non-Rust language binding, then this reference will help.
+
+### ⚠️ Caution
+_wadm and its corresponding API are under active development_. Many components are not yet written at all. This document serves as a means to solicit feedback and collaborate on API design and will likely change multiple times.
+
+## Topic Space
+The wadm API is exposed entirely as a NATS service on a topic space. All of the API operations will occur as _requests_ on a topic in the following format:
+
+```
+wadm.api.{lattice-id}.{noun}.{verb}
+```
+
+All requests and responses on this topic are encoded as JSON.
+
+If a consumer wishes to monitor events in real-time (for example, you want to be notified as soon as compensating actions are taken), then it can subscribe to the following topic:
+
+```
+wadm.evt.{lattice-id}
+```
+
+The events on this topic are JSON-encoded [CloudEvents](https://cloudevents.io/).
```

Comment:
What benefits do we get from a cloudevent envelope for wadm? I am having a hard time thinking of one right now (doesn't mean there isn't one)

Reply:
It's just as easy to create a `CloudEvent` envelope as it is to create a proprietary one. At least with a `CloudEvent` envelope, that envelope is a "well known schema", with a predictable set of fields.